### PR TITLE
Refactor bullet styles, decrease UL spacing

### PR DESCRIPTION
**Pull request by danielamorse**
## Jira

http://vjira2:8080/browse/BDS-2028

## Summary

Decreases the space between UL bullet and text, refactors how list spacing is set.

## Details

I refactored the list item spacing and the bullet position to be dynamic based on the value of a few local variables. As part of this work, I moved the UL bullet to align with the center of the OL number circle. This gives the appearance of the bullet being closer to the text.

## How to test

- Review refactored code
- OL: Verify there are no regression between [master](https://boltdesignsystem.com/pattern-lab/patterns/02-components-ol-15-ol-nested-items/02-components-ol-15-ol-nested-items.html) and [feature branch](https://boltdesignsystem-b2la3wgis.now.sh/pattern-lab/patterns/02-components-ol-15-ol-nested-items/02-components-ol-15-ol-nested-items.html)
- UL: Verify that the bullet is slightly indented on [feature branch](https://boltdesignsystem-b2la3wgis.now.sh/pattern-lab/patterns/02-components-ul-15-ul-nested-items/02-components-ul-15-ul-nested-items.html) compared to [master](https://boltdesignsystem.com/pattern-lab/patterns/02-components-ul-15-ul-nested-items/02-components-ul-15-ul-nested-items.html)
- Review above links cross-browser and verify the same

### CODE REVIEW

File: packages/components/bolt-li/src/li.scss [hunk 1]
```diff
@@ -85,9 +88,13 @@ bolt-ol {
       @include bolt-font-weight(bold);
       content: counter(li);
       position: absolute;
-      top: .25em;
-      left: bolt-spacing(small) / 2;
+      top: 0.25em;
+      left: calc(
+        #{$bolt-li-bullet-position} - (#{$bolt-ol-bullet-size} / 2)
+      ); // This is effectively the same as setting to `0`, but by using calc() it works with any given `$bolt-li-bullet-position` value
+      width: $bolt-ol-bullet-size;
```
Comment (mikemai2awesome) [inline packages/components/bolt-li/src/li.scss]: The number looks off center. Can you add the following?

```scss
height: $bolt-ol-bullet-size;
line-height: $bolt-ol-bullet-size;
```

**Comment by danielamorse** [comment]
> Can we convert that to relative value? Or just use `list-style`? Using `list-style` will make the bullets render naturally, and adapt to the font-size and font-weight.

@mikemai2awesome I think it has to be an even `px` value or sub-pixel issues occur. There is a comment in the code saying that it is intentionally using `6px`.

We could switch to `list-style` but then we can't precisely control the position of the bullet. The UL bullet would not line up perfectly with the center of the OL number when they are used together. If we're ok with that, I'd go with `list-style`. I tested it out and it looks ok to me. The bullet is just much closer to the text.

image

**Comment by danielamorse** [comment]
As discussed with @mikemai2awesome we will replace CSS-drawn UL bullets with HTML entities, and align them with the center of the numbered bullet. I'll make the update and re-request review.

**Comment by danielamorse** [comment]
@mikemai2awesome replaced css-drawn bullets with entities: [see commit](https://github.com/boltdesignsystem/bolt/pull/1759/commits/82e77ccb97513367cce4896837f6863e1350f10b).

**Comment by sghoweri** [comment]
> Ship it! 🚀

... once Travis gives the go-ahead ✅

**Comment by sghoweri** [comment]
PR was released with v2.19.0

**Pull request merged**